phoenix-core-client/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -3501,7 +3503,18 @@ public Void call() throws Exception {
                         try {
                             GLOBAL_QUERY_SERVICES_COUNTER.increment();
                             LOGGER.info("An instance of ConnectionQueryServices was created.");
-                            openConnection();
+                            CompletableFuture<Void> supplyFuture = CompletableFuture.runAsync(() -> {
+                                try {
+                                    openConnection();
+                                } catch (SQLException e) {
+                                    throw new RuntimeException(e);
+                                }
+                            });
+                            try {
+                                supplyFuture.get(120000, MILLISECONDS);
+                            } catch (TimeoutException e) {
+                                LOGGER.error("openConnection did not return any value");
```

Comment:
Why are we swallowing the exception rather than rethrowing it? Shouldn't the timeout be a failure condition rather than proceeding?